docs/work_tools/all.code_coverage.how_to_guide.md

```diff
@@ -0,0 +1,200 @@
+- `.github/codecov.yml`: Specifies Codecov-specific behaviors such as flags,
+  comments, and coverage thresholds
+
+## Coverage Configuration
+
+The `.coveragerc` file located at the repository's root defines coverage
+measurement settings:
+
+- `Source Directory`: Specifies the directory for coverage collection.
+  ```
+  [run]
+  source = helpers
+  ```
+
+- `Excluded Files`: These files are omitted from coverage reporting.
+  ```
+  [report]
+  omit =
+      */devops/compose/*
+      */helpers/test/outcomes/*/tmp.scratch/*
+  ```
+
+## GitHub Actions Workflow
+
+Coverage tests are automated via GitHub Actions -
+`.github/workflows/coverage_tests.yml`.
+
+### Workflow Schedule:
+
+- Runs daily at midnight (UTC)
+- Can be manually triggered (`workflow_dispatch`)
+- Action fails if coverage drops by `1%` (including `fast`, `slow` and
+  `superslow` tests)
+
+### Workflow Jobs:
+
+1. Fast Tests Coverage:
+
+- Runs daily or on manual trigger
+- Uploads report flagged as `fast`
+
+2. Slow Tests Coverage:
+
+- Runs daily or on manual trigger
+- Uploads report flagged as `slow`
+
+3. Superslow Tests Coverage:
+
+- Runs weekly on Monday or on manual trigger.
+- Uploads report flagged as `superslow`.
+
+4. Each job:
+
+- Generates an `XML` coverage report (`coverage.xml`)
+- Uploads reports to `Codecov` with respective flags (`fast`, `slow`,
+  `superslow`)
+- Marks CI as failed if upload errors occur (`fail_ci_if_error: true`)
+
+## Codecov Configuration (codecov.yml)
+
+Coverage flags and project-level checks are configured at -
+`.github/codecov.yml`.
+
+1. Flag Management: The `carryforward` option allows Codecov to reuse the
+   previous coverage data if no new coverage report is submitted for a specific
+   flag during a `CI` run. Setting it to true ensures continuous visibility for
+   flags that might not run on every CI cycle.
+   ```
+   flag_management:
+   default_rules:
+       carryforward: false
+   individual_flags:
+       - name: fast
+       carryforward: true
+       - name: slow
+       carryforward: true
+       - name: superslow
+       carryforward: true
+   ```
+
+2. Comment Behavior: `comment = false` disables per-line comments in GitHub PRs.
+   ```
+   comment: false
+   ```
+
+3. Coverage Status Check:
+
+- Enabled for `master` branch.
+- Compares coverage changes relatively.
+- Fails CI if total coverage decreases by 1% or more. This threshold ensures
+  that significant coverage drops do not go unnoticed, enforcing code quality
+  standards.
+  ```
+  coverage:
+  status:
+      project:
+      enabled: true
+      target_branch: master
+      comparator: relative
+      threshold: 1
+      flags:
+          - fast
+          - slow
+          - superslow
+      patch:
+      enabled: false
+  ```
+
+## Viewing Coverage Reports
+
+Coverage results for the helpers repository are accessible via Codecov.
+
+- Codecov UI link for helpers -
+  [helpers repo coverage](https://app.codecov.io/gh/causify-ai/helpers)
```

Comment:
Let's change it to a simple link -> `https://app.codecov.io/gh/causify-ai/helpers`